core/src/main/java/org/ehcache/core/statistics/TierOperationStatistic.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.core.statistics;
+
+import org.terracotta.context.ContextManager;
+import org.terracotta.context.TreeNode;
+import org.terracotta.context.annotations.ContextAttribute;
+import org.terracotta.context.query.Matcher;
+import org.terracotta.context.query.Matchers;
+import org.terracotta.context.query.Query;
+import org.terracotta.statistics.OperationStatistic;
+import org.terracotta.statistics.ValueStatistic;
+import org.terracotta.statistics.observer.ChainedOperationObserver;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.terracotta.context.query.Matchers.attributes;
+import static org.terracotta.context.query.Matchers.context;
+import static org.terracotta.context.query.Matchers.hasAttribute;
+import static org.terracotta.context.query.Matchers.identifier;
+import static org.terracotta.context.query.Matchers.subclassOf;
+import static org.terracotta.context.query.Queries.self;
+import static org.terracotta.context.query.QueryBuilder.queryBuilder;
+
+/**
+ * @author Ludovic Orban
+ */
+@ContextAttribute("this")
+public class TierOperationStatistic<S extends Enum<S>, D extends Enum<D>> implements OperationStatistic<D> {
```

Comment:
This class is heavily used by :impl and  :clustered:client.  So I assume it shouldn't be moved?